packages/shared/src/composables/use-items.ts

```diff
@@ -60,11 +60,16 @@ export function useItems(collection: Ref<string | null>, query: ComputedQuery, f
 		return Math.ceil(itemCount.value / (unref(limit) ?? 100));
 	});
 
-	let currentRequest: CancelTokenSource | null = null;
+	const existingRequests: Record<'items' | 'total' | 'filter', AbortController | null> = {
+		items: null,
+		total: null,
+		filter: null,
+	};
 	let loadingTimeout: NodeJS.Timeout | null = null;
 
 	const fetchItems = throttle(getItems, 500);
 
+	// can this be removed to prevent double fetches?
 	if (fetchOnInit) {
 		fetchItems();
 	}
```

Comment:
@rijkvanzanten is there a particular edge case where this is still required, or perhaps it's no longer applicable?

It seems like it was first added in https://github.com/directus/directus/pull/2893/files#diff-0d53067b4a1ccc2bb89ebeb8bfc7bbda1c7f94922e75668b223feaa0c64741b5 for calendar layout, to specifically opt-out calendar layout from fetching on init, with the watcher's `immediate` option based on `fetchOnInit`'s value.

However in the advanced filters PR https://github.com/directus/directus/pull/8570/files#diff-9898b066ab280a7cd156ae0d26bea7373f99c17d512e2a20d9899fe8c57ed907, the watcher is now always set to `immediate: true`, so `fetchOnInit` turned out to be making an additional redundant request. Though the request cancelation kicks in to cancel it anyways.

Reply:
Great question. Everything seems to work just fine when removing it

Reply:
>  Though the request cancelation kicks in to cancel it anyways.

Which is OK for the app, but still causes the API to use valuable resources!